flink-runtime/src/main/java/org/apache/flink/runtime/operators/coordination/SubtaskGatewayImpl.java

```diff
@@ -80,11 +86,14 @@
                                                 subtaskAccess.subtaskName());
                                 subtaskAccess.triggerTaskFailover(new FlinkException(msg, failure));
```

Comment:
I think we should guard that this method does not throw an exception. If it does, then we should fail hard. This will ensure that we don't swallow this failure as a send event failure.

Reply:
Maybe we can add a util `Runnables.assertNoException(Runnable)` that calls the `FatalExitExceptionHandler.INSTANCE`.

Reply:
Got that, we indeed should not leak the exception~ I added the util method and the check~